src/Microsoft.Identity.Client/Instance/Discovery/KnownMetadataProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Instance.Discovery
+{
+    internal class KnownMetadataProvider : IKnownMetadataProvider
+    {
+        // No need to use a ConcurrentDictionary, because the normal Dictionary is thread safe for read operations
```

Comment:
reduce comment to (// Dictionary is thread safe for read operations)...or no comment at all

Reply:
We tend to overuse ConcurrentDictionary in MSAL and so far I haven't seen a single instance where we use it correctly (hint: dict[key] = value is an incorrect way). So if we ever hit concurrency problems and a developer decides to transform this to a ConcurrentDictionary, hopefully they will heed my warning :D

Reply:
or fail trying. lol.

Reply:
But you ARE writing to this s_knownEntries dictionary on line 67.  So this needs to be a concurrentdictionary or that method needs to be locked to be threadsafe.